### PR TITLE
Rewrite header param BBEs

### PR DESCRIPTION
**Pull request by TharmiganK**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Review by praneesha — APPROVED** [review]

File: examples/http-header-param/http_header_param.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
...
+The `http` resource method allows you to access the inbound request header value as a resource argument. The header value can be either accessed as a single value or, an array of values.
+The value of a given header can be accessed as a resource method argument with the `@http:Header` annotation. The header name is resolved by the variable name unless the header name is specified in the `name` field of the annotation. The header parameter can be optional and support the basic types such as `string`, `int`, `float`, `decimal`, and `boolean`, and the `array` types of these basic types. Unless the type is optional, the request will be responded with a 400 Bad Request in the absence of the mentioned header.
```
Comment (TharmiganK) [inline examples/http-header-param/http_header_param.md:4]:
```suggestion
The value of a given header can be accessed as a resource method argument with the `@http:Header` annotation. The header name is resolved by the variable name unless the header name is specified in the `name` field of the annotation. The header parameter can be optional and support the basic types such as `string`, `int`, `float`, `decimal`, and `boolean`, and the `array` types of these basic types. Unless the type is optional, a `400 Bad Request` response will be sent to the client in the absence of the mentioned header.
```

**Review by praneesha — APPROVED** [review]

**Review by praneesha — APPROVED** [review]

File: examples/http-client-header-parameter/http_client_header_parameter.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # HTTP client - Header parameter
 
-The http module provides support for sending outbound request headers as resource method arguments. The headers can be provided as `map<string|string[]>` to the resource method. 
+The `http:Client` supports sending outbound request headers along with the request payload.
+This headers can be passed as an argument in the client resource method call. The headers should be provided as a `map` where the keys represent the header names and the entries represent the header values. The header values can be `string` or `string[]`.
+Use these headers to provide information about the request context, so that the server can tailor the response.
```
Comment (shafreenAnfar) [inline examples/http-client-header-parameter/http_client_header_parameter.md:5]: Use this when you want to send additional headers as part of the request.

File: examples/http-header-param/http_header_param.md [hunk 3]
```diff
@@ -1,6 +1,9 @@
...
+The `http` resource method allows you to access the inbound request header value as a resource argument. The header value can be either accessed as a single value or, an array of values.
+The value of a given header can be accessed as a resource method argument with the `@http:Header` annotation. The header name is resolved by the variable name unless the header name is specified in the `name` field of the annotation. The header parameter can be optional and support the basic types such as `string`, `int`, `float`, `decimal`, and `boolean`, and the `array` types of these basic types. Unless the type is optional, a `400 Bad Request` response will be sent to the client in the absence of the mentioned header.
+Use this annotated argument in the resource method to obtain the header value/values of a given header. 
+Furthermore, the values of a set of headers can be retrieved using a `record` type annotated with `@http:Header`. Here, the header names are resolved by the record field names. In addition, more header manipulations can be done via the `http:Headers` header object, which can also be accessed as a resource method argument without using the annotation.
```
Comment (shafreenAnfar) [inline examples/http-header-param/http_header_param.md:7]: The `@http:header` annotation allows reading header values from the request. The annotation can be used to annotate a given resource parameter. The name of the parameter must match the name of the header. If there is a mismatch the header name must be given in the annotation configuration. The resource parameter can be a simple type or array type. I.e. `string version`, `string[] versions`. If there are many headers to read, a record type can be used as the parameter. Unless the parameter is optional (i.e. string version = “v1”), a `400 Bad Request` response is sent to the client in the absence of the mapping header.

Comment (shafreenAnfar) [reply examples/http-header-param/http_header_param.md:7]: Lets try to make the descriptions as concise as possible.

Comment (shafreenAnfar) [reply examples/http-header-param/http_header_param.md:7]: Let's not mention about http:Headers as it is not friendly with openapi generation.

Comment (TharmiganK) [reply examples/http-header-param/http_header_param.md:4]: Fixed via [519200e](https://github.com/ballerina-platform/ballerina-distribution/pull/3763/commits/519200e50bec0a96466340292b820dc4e4810c48)

File: examples/http-client-header-parameter/http_client_header_parameter.md (hunk 2 above)
Comment (TharmiganK) [reply examples/http-client-header-parameter/http_client_header_parameter.md]: Fixed via [519200e](https://github.com/ballerina-platform/ballerina-distribution/pull/3763/commits/519200e50bec0a96466340292b820dc4e4810c48)

File: examples/http-client-header-parameter/http_client_header_parameter.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # HTTP client - Header parameter
 
-The http module provides support for sending outbound request headers as resource method arguments. The headers can be provided as `map<string|string[]>` to the resource method. 
+The HTTP client supports sending outbound request headers along with the request payload. These headers can be passed as an argument in the client resource method call. The headers should be provided as a `map` where the keys represent the header names and the entries represent the header values. The header values can be `string` or `string[]`. Use this when you want to send additional headers as part of the request.
```
Comment (ayeshLK) [inline examples/http-client-header-parameter/http_client_header_parameter.md:3]:
```suggestion
The `http:Client` supports sending outbound request headers along with the request payload. These headers can be passed as an argument in the client resource method call. The headers should be provided as a `map` where the keys represent the header names and the entries represent the header values. The header values can be `string` or `string[]`. Use this when you want to send additional headers as part of the request.
```

Comment (TharmiganK) [reply examples/http-client-header-parameter/http_client_header_parameter.md:3]: In other BBEs we have used like `The HTTP client`. So which should be followed?
@shafreenAnfar @chamil321

Comment (chamil321) [reply examples/http-client-header-parameter/http_client_header_parameter.md:3]: +1 to quote it

**Review by chamil321 — APPROVED** [review]

**Pull request merged**